test/WebJobs.Script.Tests/EndToEndTimeoutTests.cs

```diff
@@ -0,0 +1,213 @@
+             });
+        }
+
+        private async Task RunTokenTest(string scenario, Action<IList<string>> verify)
+        {
+            string functionName = "TimeoutToken";
+            TestHelpers.ClearFunctionLogs(functionName);
+
+            using (var manager = await CreateAndStartScriptHostManager("CSharp", functionName, TimeSpan.FromSeconds(3)))
+            {
+                Dictionary<string, object> arguments = new Dictionary<string, object>
+                {
+                    { "input", scenario },
+                };
+
+                var ex = await Assert.ThrowsAsync<OperationCanceledException>(() => manager.Instance.CallAsync(functionName, arguments));
+
+                Assert.True(manager.IsTimeout);
+                var logs = await TestHelpers.GetFunctionLogsAsync(functionName);
+                verify(logs);
+            }
+        }
+
+        [Fact]
+        public async Task StopHost_DoesNotThrowTimeout_CSharp()
+        {
+            string functionName = "TimeoutSync";
+            TestHelpers.ClearFunctionLogs(functionName);
+
+            var manager = await CreateAndStartScriptHostManager("CSharp", functionName, TimeSpan.FromMinutes(5));
+            string data = Guid.NewGuid().ToString();
+
+            Dictionary<string, object> arguments = new Dictionary<string, object>
+                {
+                    { "input", data },
+                };
+
+            Task t = manager.Instance.CallAsync(functionName, arguments);
+
+            await TestHelpers.Await(() =>
+            {
+                var inProgressLogs = TestHelpers.GetFunctionLogsAsync(functionName, throwOnNoLogs: false).Result;
+                return inProgressLogs?.Any(l => l.Contains(data)) ?? false;
+            });
+
+            // now stop the manager
+            manager.Stop();
+            manager.Dispose();
+
+            var ex = await Assert.ThrowsAsync<TaskCanceledException>(async () => await t);
+
+            await TestHelpers.Await(() =>
+            {
+                var inProgressLogs = TestHelpers.GetFunctionLogsAsync(functionName, throwOnNoLogs: false).Result;
+                bool completedTimeout = inProgressLogs?.Any(l => l.Contains("Function completed (Failure: Host is Stopping")) ?? false;
+                Assert.False(inProgressLogs.Any(l => l.ToLowerInvariant().Contains("timeout")));
+                return completedTimeout;
+            });
+
+            Assert.False(manager.IsTimeout);
+        }
+
+        protected async Task RunTimeoutTest(string scriptLang, string functionName)
+        {
+            TestHelpers.ClearFunctionLogs(functionName);
+            CloudQueue queue = GetQueueReference();
+
+            using (var manager = await CreateAndStartScriptHostManager(scriptLang, functionName, TimeSpan.FromSeconds(3)))
+            {
+                string testData = Guid.NewGuid().ToString();
+                await queue.AddMessageAsync(new CloudQueueMessage(testData));
```

Comment:
>await queue.AddMessageAsync(new CloudQueueMessage(testData)); [](start = 16, length = 61)

These tests aren't very clean with a queue since a Timeout immediately puts the message back on the queue, resulting in some duplicate logging before the test exits.

Mathew's just fixed manualTrigger for scripts, so I'll swap this out in my next commit.